mobile/src/components/courses/courses.tsx

```diff
@@ -49,20 +67,59 @@ const Courses: FC = (): ReactElement => {
   };
 
   const handleSearch = (search: string): void => {
-    dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    setSearchValue(search);
+  };
+
+  const handleCategorySelect = (id: number): void => {
+    if (activeCategoryId !== id) {
+      setActiveCategoryId(id);
+    } else {
+      setActiveCategoryId(null);
+    }
   };
 
+  useEffect(() => {
+    if (activeCategoryId) {
+      dispatch(categoryActions.getCategoryById({ id: activeCategoryId }));
+    } else {
+      dispatch(categoryActions.clearCategory());
+    }
+  }, [activeCategoryId]);
+
+  useEffect(() => {
+    if (courseCategory) {
+      payload.categoryKey = courseCategory.key;
```

Comment:
payload should be inside useEffect, I meant that you need to set payload conditionally and call dispatch only once